File: Source/Baseline/Iti.Email/EmailRecordDto.cs
using System;
using Iti.Baseline.Core.DTOs;
using Iti.Baseline.Identities;

namespace Iti.Baseline.Email
{
    public class EmailRecordDto : IDto
    {
        public EmailRecordId Id { get; set; }

        public NotificationId NotificationId { get; set; }

        public EmailStatus Status { get; set; }

        public DateTimeOffset? SentUtc { get; set; }

        public string ToAddress { get; set; }

        public string Subject { get; set; }

        public string Body { get; set; }

        public int RetryCount { get; set; }
        public DateTimeOffset? NextRetryUtc { get; set; }
    }
}

File: TypeScript/iti-baseline-types/EmailRecordDto.ts
import { NotificationId } from './NotificationId'
import { EmailRecordId } from './EmailRecordId'
import { EmailStatus } from './EmailStatus'

export const EmailRecordDtoTypeName = 'EmailRecordDto'
export interface EmailRecordDto  { 
    id: EmailRecordId
    notificationId: NotificationId
    status: EmailStatus
    sentUtc?: Date | null
    toAddress: string
    subject: string
    body: string
    retryCount: number
    nextRetryUtc?: Date | null
}
